catalog/registry/catalog-registry-policyplugin/src/main/java/ddf/catalog/registry/policy/RegistryPolicyPlugin.java

```diff
@@ -0,0 +1,317 @@
+     */
+    public void setRegistryBypassPolicyStrings(List<String> permStrings) {
+        this.registryBypassPolicyStrings = permStrings;
+        parsePermissionsFromString(permStrings, bypassAccessPolicy);
+    }
+
+    /**
+     * Parses a string  array representation of permission attributes
+     *
+     * @param permStrings String array of permissions to parse
+     * @param policy      The policy map to put the parsed permissions in
+     */
+    private void parsePermissionsFromString(List<String> permStrings,
+            Map<String, Set<String>> policy) {
+        policy.clear();
+        if (permStrings != null) {
+            for (String perm : permStrings) {
+                String[] parts = perm.split("=");
+                if (parts.length == 2) {
+                    String attributeName = parts[0];
+                    String attributeValue = parts[1];
+                    policy.put(attributeName, Collections.singleton(attributeValue));
+                }
+            }
+        }
+    }
+
+    @Override
+    public PolicyResponse processPreCreate(Metacard input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+        HashMap<String, Set<String>> operationPolicy = new HashMap<>();
+        if (Requests.isLocal(properties) && input.getContentTypeName() != null
+                && input.getContentTypeName()
+                .startsWith(RegistryObjectMetacardType.REGISTRY_METACARD_TYPE_NAME)) {
+            if (isRegistryDisabled()) {
+                operationPolicy.putAll(bypassAccessPolicy);
+            } else {
+                operationPolicy.putAll(writeAccessPolicy);
+            }
+        }
+        return new PolicyResponseImpl(operationPolicy, new HashMap<>());
+    }
+
+    @Override
+    public PolicyResponse processPreUpdate(Metacard input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+        HashMap<String, Set<String>> operationPolicy = new HashMap<>();
+        if (Requests.isLocal(properties) && input.getContentTypeName() != null
+                && input.getContentTypeName()
+                .startsWith(RegistryObjectMetacardType.REGISTRY_METACARD_TYPE_NAME)) {
+            if (isRegistryDisabled()) {
+                operationPolicy.putAll(bypassAccessPolicy);
+            } else {
+                operationPolicy.putAll(writeAccessPolicy);
+            }
+        }
+        return new PolicyResponseImpl(operationPolicy, new HashMap<>());
+    }
+
+    @Override
+    public PolicyResponse processPreDelete(String attributeName, List<Serializable> attributeValues,
+            Map<String, Serializable> properties) throws StopProcessingException {
+        HashMap<String, Set<String>> operationPolicy = new HashMap<>();
+        if (Requests.isLocal(properties) && catalogRegistryEntriesIncluded(attributeName,
+                attributeValues)) {
+            if (isRegistryDisabled()) {
```

Comment:
Might be worth pulling this into it's own method since it's shared between `processPreCreate`, `processPreUpdate`, and `processPreDelete`.